### PR TITLE
Task 'run' should allow arguments and options.

### LINKED ISSUE #111: Allow parameters in command definitions in runner.yml
In #25 the option was added to define commands inside the runner.yml(.dist) file.

It would be nice if we could also add parameters to those commands.

Currently this has to be defined like so, by specifying the full cli command as a single string:

```yml
commands:
  efd:build-themes:
  - "${runner.bin_dir}/run efd:npm-build --npm-dir=path/to/npm/root"
```

With support for parameters / arguments, it could look like this:

```yml
commands:
  efd:build-themes:
  -
    task: "run"
    command: "efd:npm-build"
    arguments: {npm-dir: "path/to/npm/root"}
```

(The example is inspired from this, https://github.com/ec-europa/ecdc-vaccine-reference/blob/6cf99aec072e0b6616cc6a0967be692692c53103/src/TaskRunner/Commands/EcdcBuildCommands.php#L71)

### PR DESCRIPTION
**Pull request by claudiu-cristea**
Fixes #111

### CODE REVIEW

File: tests/Tasks/CollectionFactoryTest.php [hunk 1]
```diff
@@ -74,6 +73,31 @@ public function testProcessPhpTask($type, $override, $destinationExists, $source
...
+        $tasks = [
+          [
+              'task' => 'run',
+              'command' => 'custom:test',
+              'options' => [
+                  'filepath' => $filePath,
+                  'content' => __METHOD__,
+              ],
+          ],
+        ];
```
[the remark this answers is not in the document]
Comment (pfrenssen) [reply tests/Tasks/CollectionFactoryTest.php:91]: Hmmm I tried to move the `filepath` to an argument but then the command does not execute any more. I did not have time to look deeper, but it seems none of the other commands are using arguments either?

File: tests/custom/src/TaskRunner/Commands/TestCommands.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace My\Custom\TaskRunner\Commands;
+
+use OpenEuropa\TaskRunner\Commands\AbstractCommands;
+use Symfony\Component\Console\Input\InputOption;
+
+/**
+ * Class TestCommands.
+ */
```
Comment (pfrenssen) [inline tests/custom/src/TaskRunner/Commands/TestCommands.php:10]: Maybe  `Commands for testing TaskRunner.` ?

File: tests/Tasks/CollectionFactoryTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (pfrenssen) [reply tests/Tasks/CollectionFactoryTest.php]: OK it seems that even in the current PR the command doesn't execute, the test is failing:

```
1) OpenEuropa\TaskRunner\Tests\Tasks\CollectionFactoryTest::testRunTask
file_get_contents(/test/task-runner/tests/sandbox/test-file.txt): failed to open stream: No such file or directory
```

File: tests/custom/src/TaskRunner/Commands/TestCommands.php (hunk 2 above)
Comment (claudiu-cristea) [reply tests/custom/src/TaskRunner/Commands/TestCommands.php]: Fixed.

File: tests/Tasks/CollectionFactoryTest.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (claudiu-cristea) [reply tests/Tasks/CollectionFactoryTest.php]: Thank you for finding out the source of test failure. Foxed the test and tested also the command arguments.

**Comment by claudiu-cristea** [comment]
Fixed remarks & test.

**Pull request merged**